### ISSUE TITLE
What Would You Like Added to Survey Module?

### ISSUE THREAD

**Issue by MaiklT** [DNNCommunity/DNN.Survey#4]
In 2011, a thread in the DNN forums started about what should be added to the Survey module. There were some nice suggestions, but nothing has happened...

Therefore, I give it another try:

* Single choice grid with "from-to" selection
* Single choice grid with options in the header
* Multiple choice (checkboxes) grid with options in the header
* Free text answers, as well as single-line and multi-line textboxes
* Headings for different chapters
* Labels for explanations etc.
* Export of results to xlsx (or csv at least)
* (Not so important:) Improved templates and styling capabilities (themes)
* (Important, but propably not realistic): ML support
* Multi-page capabilities (one page after the other has to be answered)
* Tabbed surveys (user cab jump between pages as he wants)

See also [DNN Forum](https://www.dnnsoftware.com/forums/forumid/51/threadid/426727/scope/posts/threadpage/1)

**Comment by valadas** [comment]
Those are great ideas, I have invested a lot of time to make most of the most popular core modules compatible with Dnn 9.2, I then focused on trying to iron out bugs. This particular module was not amongst the most populars and it currently still is on codeplex. If there is any interest in submitting pull requests for some of those features, let me know and I will move it to this repository, unfortunatelly time being a limited resource, I still had no time to get to this one. If there is interest, it would be good to separate each of these changes/improvements into individual issues.

**Comment by MaiklT** [comment]
Daniel,

thanks for the answer. The real problem is that time IS a very limited resource. The other thing is that the client who needed all of these changed to another product, so my need for that is not really big.

Anyway: I'll download the stuff from CodePlex and try to translate it to C# and make it DNN 9 compatible as it is. We can then talk about further features and make it better...

Happy DNNing!
Michael

**Comment by MaiklT** [comment]
Daniel,

I created single feature requests for the most important requests (at least the ones that are imortant in my eyes :-)).

I already started the project (https://github.com/MaiklT/DNN.Survey/tree/develop) - hope I can start a pull request soon. I already implemented the text box answer, and some other nice new features, as sort by drag and drop and HTML5 charts created with chartjs (https://www.chartjs.org/).

Happy DNNing!
Michael
